AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -177,16 +173,16 @@ - (void)generatePlaceholderImage
 - (void)imageAtTime:(CMTime)imageTime completionHandler:(void(^)(UIImage *image))completionHandler
 {
   ASPerformBlockOnBackgroundThread(^{
-    ASDN::MutexLocker l(_videoLock);
+    AVAsset *asset = self.asset;
```

Comment:
Why is it essential to eliminate this lock? Isn't it possible that the asset could change before the next call, which could mean that code which intends to cancel loading of the media may not be able to? I am very unfamiliar with this area of the API, and so I am not sure if it is plausible to initiate an operation here that may require downloading a significant amount of information from the network, and thus should be cancelable if there is a race with setAsset:nil on another thread.